src/main/java/fi/aalto/cs/apluscourses/utils/FunctionUtil.java

```diff
@@ -0,0 +1,13 @@
+package fi.aalto.cs.apluscourses.utils;
+
+import java.util.function.IntPredicate;
+
+public class FunctionUtil {
```

Comment:
we probably have more custom predicates across the code, so far they were located close to where used, do you think we shall extract them too :question:

Reply:
No. Already this may be an overkill here. The logic where this was used was so complex that I tried to make it easier to read by putting parts of it in separate functions. I suggest that if, after a few sprints, this class still exists only for that single purpose, let's inline it. But perhaps there will be more methods/uses before that. So let's see.

Reply:
In this case I would actually favor having the predicate inline in the code, `y -> y <= x` is very readable and one doesn't have to check that `lessThanOrEqualTo` doesn't have any surprises

Reply:
Oops posted my above comment before refreshing the page so your reply wasn't visible, your plan of coming back to this later sounds good :+1: